flowmachine/Pipfile

```diff
@@ -34,6 +34,7 @@ pytest-cov = "*"
 pytest-testmon = "*"
 asynctest = "*"
 cachey = "*"
+approvaltests = "*"
```

Comment:
Could you explain the idea behind this and how it differs from other approaches to testing?

Reply:
The main idea is that you produce some output and ask approvaltests to verify it. The first time the test runs you are asked to approve the output, and approaltests then stores it in a file. On subsequent test runs it simply compares the new output to the previous one. The test passes if the output hasn't changed, otherwise it fails.

This is basically the same idea as with usual tests and assert statements. The main advantage is in situations where the output is large (e.g. a full dataframe or a large SQL string, or any other large object). This is where approvaltests takes a lot of the scaffolding and boilerplate away from you because it presents you with a diff that you can quickly inspect visually and then either approve or reject.

A lot of our existing tests in flowmachine only check a few hand-selected values. This is usually ok, but makes it easy to miss subtle differences during refactoring. I wanted to ensure that the full output is _exactly_ as it was before, or if there are any changes to see exactly what they are (both in terms of the generated SQL and the resulting dataframe), which approvaltests makes very easy.